File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.java
import java.util.*;


class Solution {
    
    public Map<String, Integer> hm;
    
    public String solution(int[] numLog) {
        String answer = "";
        hm = new HashMap<>();
        
        hm.put("w", 1);
        hm.put("s", -1);
        hm.put("d", 10);
        hm.put("a", -10);
        
        for(int i = 1; i < numLog.length; i++) {
            int prev = numLog[i - 1];
            int curr = numLog[i];
            
            for(Map.Entry<String, Integer> entry : hm.entrySet()) {
                int res = prev + entry.getValue();
                
                if(res == curr) {
                    answer += entry.getKey();
                    break;
                }
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.js
function solution(numLog) {
    var answer = '';
    const order = {
        'w': 1,
        's': -1,
        'd': 10,
        'a': -10
    }
    
    for(let i = 1; i < numLog.length; i++) {
        const prev = numLog[i - 1];
        const curr = numLog[i];
        
        for(let [key, value] of Object.entries(order)) {
            const res = prev + value;
            
            if(res === curr) {
                answer += key;
                break;
            }
        }
    }
    
    return answer;
}